enterprise/server/crypter_service/crypter_service.go

```diff
@@ -6,53 +6,387 @@ import (
 	"crypto/rand"
 	"crypto/sha256"
 	"errors"
+	"flag"
 	"fmt"
 	"io"
 	"strings"
+	"sync"
+	"time"
 
 	"github.com/buildbuddy-io/buildbuddy/server/environment"
 	"github.com/buildbuddy-io/buildbuddy/server/interfaces"
 	"github.com/buildbuddy-io/buildbuddy/server/remote_cache/digest"
 	"github.com/buildbuddy-io/buildbuddy/server/tables"
+	"github.com/buildbuddy-io/buildbuddy/server/util/log"
+	"github.com/buildbuddy-io/buildbuddy/server/util/retry"
 	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+	"github.com/jonboulle/clockwork"
```

Comment:
nit: I recall at one point I pulled in a lib like this and after some discussion w/ Tyler I added `timeutil.Clock` and `testclock` utils. I don't have a super strong opinion but we might want to use those just for consistency, to avoid introducing this direct dep

Reply:
I'm relying on some additional functionality from the clockwork library, notably `time.After` which is slightly more complicated. 

I understand his reasoning for not wanting to add a dep for a simple use like `time.Now` but now that we have more complex needs I think it's reasonable to pull in the `clockwork` dep instead of implementing it ourselves. 

To make things more consistent, I can go ahead and replace the use of timeutil.Clock with clockwork (in a separate PR). How do you feel about that?